### DISCUSSION TITLE
Valgrind uninitialized bytes

### DISCUSSION THREAD

**Discussion by supahero1** [microsoft/msquic#3503] [Q&A]
Hello,

Lately, while using Valgrind to test my stuff, I've started being increasingly
annoyed at its messages about MsQuic, involving socket file descriptors I reckon:

Syscall param socketcall.setsockopt(optval) points to uninitialised byte(s)
&
Syscall param sendmsg(msg.msg_control) points to uninitialised byte(s)

So first of all, do you guys know about this? Do I have to worry about that?

And, less important but still, is there some flag I can enable during
compilation that will eliminate these messages, something like zeroing all
underlying buffers in debug mode or so? I've compiled the library without
any special flags, just straightforwardly did what was said in BUILD.md.

Cheers,
supahero1

**Comment by supahero1** [comment]
Actually, my bad. The sendmsg one is about "CxPlatSendDataAlloc" heap allocation, which is not sfd related. I don't know what the first one is about.

**Reply by nibanks** [reply]
If you can open an issue with all the details you have we can try to get them fixed. I don't expect they're anything real to worry about. Probably just Valgrind being paranoid, but I'd be happy to "fix" them.

**Reply by Anas198900** [reply]
> Actually, my bad. The sendmsg one is about "CxPlatSendDataAlloc" heap allocation, which is not sfd related. I don't know what the first one is about.

**Reply by Anas198900** [reply]
> If you can open an issue with all the details you have we can try to get them fixed. I don't expect they're anything real to worry about. Probably just Valgrind being paranoid, but I'd be happy to "fix" them.
